src/schema/feeds.ts

```diff
@@ -1021,14 +1041,19 @@ const applyFeedPagingWithPin = (
   return newBuilder;
 };
 
-const getFeedSettings = async (
-  ctx: Context,
-  info: GraphQLResolveInfo,
-): Promise<GQLFeedSettings> => {
+const getFeedSettings = async ({
+  ctx,
+  info,
+  feedId,
+}: {
+  ctx: Context;
+  info: GraphQLResolveInfo;
+  feedId: string;
+}): Promise<GQLFeedSettings> => {
   const res = await graphorm.query<GQLFeedSettings>(ctx, info, (builder) => {
     builder.queryBuilder = builder.queryBuilder.andWhere(
-      `"${builder.alias}".id = :userId AND "${builder.alias}"."userId" = :userId`,
-      { userId: ctx.userId },
+      `"${builder.alias}".id = :feedId AND "${builder.alias}"."userId" = :userId`,
```

Comment:
Don't you need to do the `feedId || userId` here as well?

Reply:
Yeah, but I did it by modfying the `getFeedSettings` to accept `feedId` as well so it is passed correctly from the outside.